src/librustdoc/html/format.rs

```diff
@@ -460,7 +460,8 @@ impl fmt::Display for clean::Type {
                 f.write_str(name)
             }
             clean::ResolvedPath{ did, ref typarams, ref path } => {
-                try!(resolved_path(f, did, path, false));
+                // Paths like Self::Output should be rendered with all segments
```

Comment:
I believe these sort of associated-type paths exist are possible for any type, that is, they can start with something other than `Self`. I have no idea how to detect such things.